src/com/owncloud/android/utils/FileStorageUtils.java

```diff
@@ -184,15 +184,7 @@ public static RemoteFile fillRemoteFile(OCFile ocFile){
         
         Collections.sort(files, new Comparator<OCFile>() {
             public int compare(OCFile o1, OCFile o2) {
-                if (o1.isFolder() && o2.isFolder()) {
-                    Long obj1 = o1.getModificationTimestamp();
-                    return val * obj1.compareTo(o2.getModificationTimestamp());
-                }
-                else if (o1.isFolder()) {
-                    return -1;
-                } else if (o2.isFolder()) {
-                    return 1;
-                } else if (o1.getModificationTimestamp() == 0 || o2.getModificationTimestamp() == 0){
+                if (o1.getModificationTimestamp() == 0 || o2.getModificationTimestamp() == 0){
```

Comment:
I don't really like that if there is no timestamp we artificially making it equal with everything. This leads to unexpected behavior when we do have multiple zeros in collection.
I would rather not take any special care of this and boil down compare method to something like this:
```java
public int compare(OCFile o1, OCFile o2) {
    return val * (o1.getModificationTimestamp() - o2.getModificationTimestamp());
}
```